source/jobs/JobEngine.cpp

```diff
@@ -139,18 +139,37 @@ void JobEngine::exec_action(PlainJobDocument::JobAction action, const std::strin
     string command;
     if (action.type == PlainJobDocument::ACTION_TYPE_RUN_HANDLER)
     {
-        // build command
-        try
+        // build command for shell command
+        if (action.name == "shell-command")
```

Comment:
Nit. You can prolly put the try-catch around the `if-else` since the catch handler is the same.

Reply:
addressed